src/meta/src/stream/scheduler.rs

```diff
@@ -202,10 +203,13 @@ where
             // Normal fragment
 
             // Find out all the hash parallel units in the cluster.
-            let parallel_units = self
+            let mut parallel_units = self
                 .cluster_manager
                 .list_parallel_units(Some(ParallelUnitType::Hash))
                 .await;
+            // FIXME(Kexiang): select appropriate parallel_units, currently only support
+            // `parallel_degree < parallel_units.size()`
+            parallel_units.truncate(parallel_degree as usize);
```

Comment:
Did you test the case where parallel unit number is larger than limit?

This looks strange to me. How many actors can be scheduled is determined by how many actors are generated. If they don't match with each other, the stream graph will simply stuck.

Reply:
Okay, my fault, no problem with this part. But to ensure scheduling is fair, better to randomly pick some or let the scheduler aware of parallelism before calling this function.

Reply:
Will consider optimizing it later. Currently, it's a little bit hard to implement. The reason is that we have chained nodes that require the downstream actors to share the same Parallel Unit set with the upstream actors, so we cannot randomly choose Parallel Units.